controllers/platformnetwork_controller.go

```diff
@@ -407,19 +471,31 @@ func (r *PlatformNetworkReconciler) FindExistingAddressPool(client *gophercloud.
 // the network resource since at the system level these are two independent
 // resources.
 func (r *PlatformNetworkReconciler) ReconcileAddressPool(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) error {
-	pool, err := r.FindExistingAddressPool(client, instance)
+	var pool *addresspools.AddressPool
```

Comment:
is there a specific reason to explicitly call out the data type - *addresspools.AddressPool ?

Reply:
Previously there was an implementation to `FindAddressPoolByNetworkType` which is merged with `FindExistingAddressPool` currently.
There was if and else statement used at the very beginning to obtain the pool from these methods depending upon network type, initializing the vars inside if-else block would mean scope of the var is limited to only that block which wouldn't work.